trackmania_rl/buffer_utilities.py

```diff
@@ -77,9 +77,10 @@ def buffer_collate_function(batch):
         )
     )
 
+    rng = np.random.default_rng(42)
     temporal_mini_race_current_time_actions = (
         np.abs(
-            np.random.randint(
+            rng.integers(
```

Comment:
Given that rng is initialized just above, I believe we will always receive the same random numbers.


Obligatory xkcd: https://xkcd.com/221/

I'd prefer to have a single random number generator, initialized once at the beginning.

Reply:
I changed the code in the PR to create 1 RNG and then spawn independent child RNG from it as described in
https://blog.scientific-python.org/numpy/numpy-rng/
at the bottom.

Had to refactor buffer_utilities a bit to avoid bouncing adding an additional parameter everywhere. Added a class BufferUtil there to hold the RNG instead.